game/game/playercontrol.cpp

```diff
@@ -855,13 +868,17 @@ void PlayerControl::processPickLock(Npc& pl, Interactive& inter, KeyCodec::Actio
   }
 
 void PlayerControl::processLadder(Npc& pl, Interactive& inter, KeyCodec::Action key) {
-  if(key==KeyCodec::Back) {
-    pl.setInteraction(nullptr);
+  if(key==KeyCodec::Back && !isPressed(KeyCodec::Forward)) {
+    inter.nextLadderMove(false,true);
     return;
     }
-
-  if(key==KeyCodec::Forward) {
-    inter.nextState(pl);
+  if(key==KeyCodec::Forward && !isPressed(KeyCodec::Back)) {
+    inter.nextLadderMove(false);
```

Comment:
It's probably better to stick with `nextState` notation, as regular MOBSI are also allowed to have multiple states.
Not in use in vanilla, but can be in mods.

Reply:
Sure